src/main/java/spireMapOverhaul/zones/manasurge/ui/campfire/CampfireEnchantEffect.java

```diff
@@ -0,0 +1,109 @@
+package spireMapOverhaul.zones.manasurge.ui.campfire;
+
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.math.Interpolation;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.CardGroup;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.helpers.ImageMaster;
+import com.megacrit.cardcrawl.rooms.AbstractRoom;
+import com.megacrit.cardcrawl.rooms.CampfireUI;
+import com.megacrit.cardcrawl.rooms.RestRoom;
+import com.megacrit.cardcrawl.vfx.AbstractGameEffect;
+import com.megacrit.cardcrawl.vfx.cardManip.ShowCardBrieflyEffect;
+import spireMapOverhaul.zones.manasurge.ManaSurgeZone;
+
+import java.util.Iterator;
+
+public class CampfireEnchantEffect extends AbstractGameEffect {
+    private static final float DUR = 1.5F;
+    private boolean openedScreen = false;
+    private Color screenColor;
+
+    public CampfireEnchantEffect() {
+        this.screenColor = AbstractDungeon.fadeColor.cpy();
+        this.duration = DUR;
+        this.screenColor.a = 0.0F;
+        AbstractDungeon.overlayMenu.proceedButton.hide();
+    }
+
+    public void update() {
+        if (!AbstractDungeon.isScreenUp) {
+            this.duration -= Gdx.graphics.getDeltaTime();
+            this.updateBlackScreenColor();
+        }
+
+        Iterator var1;
```

Comment:
Please don't use whilter iterator loops if you don't have to, a simple for each works just as well

Reply:
Sorted. I copied from base game code so I'll do that instead.

Reply:
Not quite, you looked at decompiled game through intellij, the compiler is notoriously bad at for each loops and always does while iterator loops.

Base game actually also uses for each

Reply:
Ah, good to know that's just intellij jank